src/codegen/mod.rs

```diff
@@ -3962,6 +3966,50 @@ impl CodeGenerator for ObjCInterface {
                         }
                     };
                     result.push(impl_trait);
+                    for protocol_id in parent.conforms_to.iter() {
+                        if !protocol_set.contains(protocol_id) {
+                            protocol_set.insert(*protocol_id);
+                            let protocol_name = ctx.rust_ident(
+                                ctx.resolve_type(
+                                    protocol_id.expect_type_id(ctx),
+                                )
+                                .name()
+                                .unwrap(),
+                            );
+                            let impl_trait = quote! {
+                                impl #protocol_name for #class_name { }
+                            };
+                            result.push(impl_trait);
+                        }
+                    }
+                    if !parent.is_template() {
+                        let parent_struct_name = parent.name();
+                        let child_struct_name = self.name();
+                        let parent_struct = ctx.rust_ident(parent_struct_name);
+                        let from_block = quote! {
+                            impl From<#class_name> for #parent_struct {
+                                fn from(child: #class_name) -> #parent_struct {
+                                    #parent_struct(child.0)
+                                }
+                            }
+                        };
+                        result.push(from_block);
+
+                        let try_into_block = quote! {
+                            impl std::convert::TryFrom<#parent_struct> for #class_name {
+                                type Error = String;
+                                fn try_from(parent: #parent_struct) -> Result<#class_name, Self::Error> {
+                                    let is_kind_of : bool = unsafe { msg_send!(parent, isKindOfClass:class!(#class_name))};
+                                    if is_kind_of {
+                                        Ok(#class_name(parent.0))
+                                    } else {
+                                        Err(format!("This {} is not an cannot be downcasted to {}", #parent_struct_name, #child_struct_name))
```

Comment:
`and cannot be` probably?

It'd be nice to make this a `&'static str` instead, that way you avoid `try_into` allocating, which I think is nice.